src/coreComponents/virtualElement/ConformingVirtualElementOrder1.cpp

```diff
@@ -0,0 +1,433 @@
+{
+  // Get geometry managers
+  NodeManager const & nodeManager = *mesh.getNodeManager();
+  FaceManager const & faceManager = *mesh.getFaceManager();
+  ElementRegionManager const & elementManager = *mesh.getElemManager();
+
+  // Get pre-computed maps
+  CellElementRegion const & cellRegion =
+    *elementManager.GetRegion< CellElementRegion >( regionIndex );
+  CellElementSubRegion const & cellSubRegion =
+    *cellRegion.GetSubRegion< CellElementSubRegion >( subRegionIndex );
+  CellElementSubRegion::NodeMapType const & cellToNodes = cellSubRegion.nodeList();
+  FixedOneToManyRelation const & elementToFaceMap = cellSubRegion.faceList();
+  arrayView2d< real64 const, nodes::REFERENCE_POSITION_USD > nodesCoords =
+    nodeManager.referencePosition();
+
+  // Get pre-computed geometrical properties.
+  arrayView2d< real64 const > faceCenters = faceManager.faceCenter();
+  arrayView2d< real64 const > faceNormals = faceManager.faceNormal();
+  arrayView2d< real64 const > cellCenters = cellSubRegion.getElementCenter();
+  arrayView1d< real64 const > cellVolumes = cellSubRegion.getElementVolume();
+  arraySlice1d< real64 const > cellCenter = cellCenters[cellIndex];
+  localIndex const numCellFaces = elementToFaceMap[cellIndex].size();
+  localIndex const numCellPoints = cellToNodes[cellIndex].size();
+  numSupportPoints = numCellPoints;
+
+  // Compute other geometrical properties.
+  //  - compute map used to locate local point position by global id (used in the computation of
+  //    basis functions boundary integrals.
+  map< localIndex, localIndex > cellPointsPosition;
+  for( localIndex numVertex = 0; numVertex < numCellPoints; ++numVertex )
+    cellPointsPosition.insert( std::pair< localIndex, localIndex >
+                                 ( cellToNodes( cellIndex, numVertex ), numVertex ));
+  // - compute cell diameter.
+  real64 cellDiameter = 0;
+  for( localIndex numVertex = 0; numVertex < numCellPoints; ++numVertex )
+  {
+    for( localIndex numOthVertex = 0; numOthVertex < numVertex; ++numOthVertex )
+    {
+      array1d< real64 > vertDiff( 3 );
+      LvArray::tensorOps::copy< 3 >( vertDiff, nodesCoords[cellToNodes( cellIndex, numVertex )] );
+      LvArray::tensorOps::subtract< 3 >( vertDiff,
+                                         nodesCoords[cellToNodes( cellIndex, numOthVertex )] );
+      real64 const candidateDiameter = LvArray::tensorOps::l2NormSquared< 3 >( vertDiff );
+      if( cellDiameter < candidateDiameter )
+        cellDiameter = candidateDiameter;
+    }
+  }
+  cellDiameter = LvArray::math::sqrt< real64 >( cellDiameter );
+  real64 const invCellDiameter = 1.0/cellDiameter;
+
+  // Compute basis functions and scaled monomials integrals on the boundary.
+  array1d< real64 > basisBoundaryIntegrals( numCellPoints );
+  array2d< real64 > basisTimesNormalBoundaryInt( 3, numCellPoints );
+  array2d< real64 > basisTimesMonomNormalDerBoundaryInt( numCellPoints, 3 );
+  array1d< real64 > monomBoundaryIntegrals( 4 );
```

Comment:
```suggestion
  real64 monomBoundaryIntegrals[ 4 ] = { 0.0 };
```
and in general I think you can replace (maybe in the next PR) all the `array1d` and `array2d` that have a fixed size with C-arrays.